### PR TITLE
Beta/Objects

### PR DESCRIPTION
**Pull request by joshLong145**
* Addition of internal object pkg for binding to higher level targets.
## Changes
* Item 1
* Item 2

## API Updates
following internal api is being proposed for objects. we can discuss adding an internal reference to `AppSchemaInternal` to allow for further helper methods to statically assert on an object with a provided schema. at this time the interface supports uploading and retrieving objects from `ipfs` storage.

```
type AppObjectInternal interface {
	/*
		Persists an object definition to the storage configured within the module.
	*/
	CreateObject(
		schema schemas.AppSchemaInternal,
		label string,
		object map[string]interface{}) (*ObjectUploadResult, error)

	/*
		Retrieves an object for the data store
	*/
	GetObject(cid string) (map[string]interface{}, error)
}
```

## Checklist
* [ ] Unit tests
* [ ] Documentation

## References _(optional)_

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202528620406598/1202703507037753) by [Unito](https://www.unito.io)

### CODE REVIEW

File: x/registry/module_simulation.go [hunk 1]
```diff
@@ -86,73 +86,74 @@ func (am AppModule) RegisterStoreDecoder(_ sdk.StoreDecoderRegistry) {}
...
-	simState.AppParams.GetOrGenerate(simState.Cdc, opWeightMsgDeactivateWhoIs, &weightMsgDeactivateWhoIs, nil,
-		func(_ *rand.Rand) {
-			weightMsgDeactivateWhoIs = defaultWeightMsgDeactivateWhoIs
-		},
-	)
-	operations = append(operations, simulation.NewWeightedOperation(
-		weightMsgDeactivateWhoIs,
-		registrysimulation.SimulateMsgDeactivateWhoIs(am.accountKeeper, am.bankKeeper, am.keeper),
-	))
+	/*
```
Comment (mcjcloud) [inline x/registry/module_simulation.go]: Why is this all commented out?

File: pkg/object/object.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	Code       int32
+	Definition *ObjectDefinition
+	Message    string
+}
+
+type AppObjectInternalImpl struct {
+	shell *shell.Shell
+}
+
+func New(schemaImpl schemas.AppSchemaInternal, shell *shell.Shell) AppObjectInternal {
```
Comment (mcjcloud) [inline pkg/object/object.go:35]: You're not using the value of `schemaImpl` here. You could keep this in `AppObjectInternalImpl` and then remove the need to pass it into the create method.

**Review by mcjcloud — CHANGES_REQUESTED** [review]
Two tiny comments.

This is so hype.

File: app.env.sample [hunk 3]
```diff
@@ -10,4 +10,4 @@ PATH_PREFIX=/
 COSMOS_NODE_ADDRESS=http://localhost:26656  # Address of Cosmos backend
 LOCAL_CONFIG_DIR="~/.config/sonr" # a directory in which all local data is stored
 JWT_SECRET=BOBANDALICE
-IPFS_API=:api.ipfs.sonr.ws
\ No newline at end of file
+IPFS_API_READ =api.ipfs.sonr.ws
```
Comment (ntindle) [inline app.env.sample:13]: Technically this is ipfs api write

**Review by ntindle — COMMENTED** [review]
Where are the cmd commands

File: app.env.sample (hunk 3 above)
Comment (joshLong145) [reply app.env.sample:13]: indeed it is, ill update

File: pkg/object/object.go (hunk 2 above)
Comment (joshLong145) [reply pkg/object/object.go:35]: that will work, but then each schema is 1-1 with an object.

File: x/registry/module_simulation.go (hunk 1 above)
Comment (joshLong145) [reply x/registry/module_simulation.go]: was for testing, i need to stop doing this.

File: pkg/object/object.go (hunk 2 above)
Comment (mcjcloud) [reply pkg/object/object.go:35]: Good point.

**Review by mcjcloud — APPROVED** [review]

**Pull request merged**